pkg/log/logger.go

```diff
@@ -0,0 +1,90 @@
+package log
+
+import (
+	"context"
+	"io"
+
+	saltLog "github.com/goto/salt/log"
+)
+
+type Logger interface {
+
+	// Debug level message with alternating key/value pairs
+	// key should be string, value could be anything printable
+	Debug(ctx context.Context, msg string, args ...interface{})
+
+	// Info level message with alternating key/value pairs
+	// key should be string, value could be anything printable
+	Info(ctx context.Context, msg string, args ...interface{})
+
+	// Warn level message with alternating key/value pairs
+	// key should be string, value could be anything printable
+	Warn(ctx context.Context, msg string, args ...interface{})
+
+	// Error level message with alternating key/value pairs
+	// key should be string, value could be anything printable
+	Error(ctx context.Context, msg string, args ...interface{})
+
+	// Fatal level message with alternating key/value pairs
+	// key should be string, value could be anything printable
+	Fatal(ctx context.Context, msg string, args ...interface{})
+
+	// Level returns priority level for which this logger will filter logs
+	Level() string
+
+	// Writer used to print logs
+	Writer() io.Writer
+}
+
+type CtxLogger struct {
+	log saltLog.Logger
+	key string
```

Comment:
let's make this accept multiple `keys` right away?
also, we can have context key other than `string` type as well, so we can change the type to `interface{}`

```suggestion
	log  saltLog.Logger
	keys []interface{}
```

Reply:
Since we use these keys to be printed in log, wouldn't every key be cast to `string` before printed. 🤔 so maybe something like - `keys []string`. WDYT ?

Reply:
sure, `[]string` works then

Reply:
updated.